File: CS/Scaffolding.CodeFirst/ViewModels/DepartmentContextViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.ComponentModel;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm.ViewModel;
using Scaffolding.CodeFirst.DepartmentContextDataModel;

namespace Scaffolding.CodeFirst.ViewModels {
    /// <summary>
    /// Represents the root POCO view model for the DepartmentContext data model.
    /// </summary>
    public partial class DepartmentContextViewModel : DocumentsViewModel<DepartmentContextModuleDescription, IDepartmentContextUnitOfWork> {

        const string TablesGroup = "Tables";

        const string ViewsGroup = "Views";

        /// <summary>
        /// Creates a new instance of DepartmentContextViewModel as a POCO view model.
        /// </summary>
        public static DepartmentContextViewModel Create() {
            return ViewModelSource.Create(() => new DepartmentContextViewModel());
        }

        /// <summary>
        /// Initializes a new instance of the DepartmentContextViewModel class.
        /// This constructor is declared protected to avoid undesired instantiation of the DepartmentContextViewModel type without the POCO proxy factory.
        /// </summary>
        protected DepartmentContextViewModel()
            : base(UnitOfWorkSource.GetUnitOfWorkFactory()) {
        }

        protected override DepartmentContextModuleDescription[] CreateModules() {
            return new DepartmentContextModuleDescription[] {
                new DepartmentContextModuleDescription("Courses", "CourseCollectionView", TablesGroup, GetPeekCollectionViewModelFactory(x => x.Courses)),
                new DepartmentContextModuleDescription("Departments", "DepartmentCollectionView", TablesGroup, GetPeekCollectionViewModelFactory(x => x.Departments)),
                new DepartmentContextModuleDescription("Employees", "EmployeeCollectionView", TablesGroup, GetPeekCollectionViewModelFactory(x => x.Employees)),
            };
        }



    }

    public partial class DepartmentContextModuleDescription : ModuleDescription<DepartmentContextModuleDescription> {
        public DepartmentContextModuleDescription(string title, string documentType, string group, Func<DepartmentContextModuleDescription, object> peekCollectionViewModelFactory = null)
            : base(title, documentType, group, peekCollectionViewModelFactory) {
        }
    }
}

File: VB/Scaffolding.CodeFirst/ViewModels/DepartmentContextViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.ComponentModel
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm.ViewModel
Imports Scaffolding.CodeFirst.DepartmentContextDataModel

Namespace Scaffolding.CodeFirst.ViewModels
	''' <summary>
	''' Represents the root POCO view model for the DepartmentContext data model.
	''' </summary>
	Partial Public Class DepartmentContextViewModel
		Inherits DocumentsViewModel(Of DepartmentContextModuleDescription, IDepartmentContextUnitOfWork)

		Private Const TablesGroup As String = "Tables"

		Private Const ViewsGroup As String = "Views"

		''' <summary>
		''' Creates a new instance of DepartmentContextViewModel as a POCO view model.
		''' </summary>
		Public Shared Function Create() As DepartmentContextViewModel
			Return ViewModelSource.Create(Function() New DepartmentContextViewModel())
		End Function

		''' <summary>
		''' Initializes a new instance of the DepartmentContextViewModel class.
		''' This constructor is declared protected to avoid undesired instantiation of the DepartmentContextViewModel type without the POCO proxy factory.
		''' </summary>
		Protected Sub New()
			MyBase.New(UnitOfWorkSource.GetUnitOfWorkFactory())
		End Sub

		Protected Overrides Function CreateModules() As DepartmentContextModuleDescription()
			Return New DepartmentContextModuleDescription() {
				New DepartmentContextModuleDescription("Courses", "CourseCollectionView", TablesGroup, GetPeekCollectionViewModelFactory(Function(x) x.Courses)),
				New DepartmentContextModuleDescription("Departments", "DepartmentCollectionView", TablesGroup, GetPeekCollectionViewModelFactory(Function(x) x.Departments)),
				New DepartmentContextModuleDescription("Employees", "EmployeeCollectionView", TablesGroup, GetPeekCollectionViewModelFactory(Function(x) x.Employees))
			}
		End Function



	End Class

	Partial Public Class DepartmentContextModuleDescription
		Inherits ModuleDescription(Of DepartmentContextModuleDescription)

		Public Sub New(ByVal title As String, ByVal documentType As String, ByVal group As String, Optional ByVal peekCollectionViewModelFactory As Func(Of DepartmentContextModuleDescription, Object) = Nothing)
			MyBase.New(title, documentType, group, peekCollectionViewModelFactory)
		End Sub
	End Class
End Namespace